msteams-platform/concepts/authentication/nested-authentication.md

```diff
@@ -0,0 +1,343 @@
+
+1. If no account is available, MSAL.js returns an `InteractionRequiredAuthError`. Call `publicClientApplication.acquireTokenPopup(accessTokenRequest)` to display an interactive dialog for the user. `acquireTokenSilent` can fail if the token expired or if the user didn't consent to all the requested scopes.
+
+The following code snippet shows an example to access a token:
+
+# [JavaScript](#tab/js2)
+
+```javascript
+
+  // MSAL.js exposes several account APIs, logic to determine which account to use is the responsibility of the developer
+  const account = publicClientApplication.getActiveAccount();
+
+  const accessTokenRequest = {
+  scopes: ["user.read"],
+  account: account,
+  };
+
+  publicClientApplication
+    .acquireTokenSilent(accessTokenRequest)
+    .then(function (accessTokenResponse) {
+      // Acquire token silent success
+      let accessToken = accessTokenResponse.accessToken;
+      // Call your API with token
+      callApi(accessToken);
+    })
+    .catch(function (error) {
+      //Acquire token silent failure, and send an interactive request
+      if (error instanceof InteractionRequiredAuthError) {
+        publicClientApplication
+          .acquireTokenPopup(accessTokenRequest)
+          .then(function (accessTokenResponse) {
+            // Acquire token interactive success
+            let accessToken = accessTokenResponse.accessToken;
+            // Call your API with token
+            callApi(accessToken);
+          })
+          .catch(function (error) {
+            // Acquire token interactive failure
+            console.log(error);
+          });
+      }
+      console.log(error);
+    });
+
+```
+
+# [C#](#tab/cs2)
+
+```csharp
+// MSAL.NET exposes several account APIs, logic to determine which account to use is the responsibility of the developer
+var account = publicClientApplication.GetAccountsAsync().Result.FirstOrDefault();
+var accessTokenRequest = publicClientApplication.AcquireTokenSilent(new[] { "user.read" }, account);
+
+try
+{
+    var accessTokenResponse = accessTokenRequest.ExecuteAsync().Result;
+    // Acquire token silent success
+    var accessToken = accessTokenResponse.AccessToken;
+    // Call your API with token
+    CallApi(accessToken);
+}
+catch (MsalUiRequiredException ex)
+{
+    // Acquire token silent failure, and send an interactive request
+    try
+    {
+        var accessTokenResponse = publicClientApplication.AcquireTokenInteractive(new[] { "user.read" }).ExecuteAsync().Result;
+        // Acquire token interactive success
+        var accessToken = accessTokenResponse.AccessToken;
+        // Call your API with token
+        CallApi(accessToken);
+    }
+    catch (Exception innerEx)
+    {
+        // Acquire token interactive failure
+        Console.WriteLine(innerEx);
+    }
+}
+catch (Exception ex)
+{
+    Console.WriteLine(ex);
+}
+```
+
```

Comment:
Remove the C# code for all the code snippets.

Reply:
@Wajeed-msft after I remove all the C# code snippets, please let me know if should remove the C# code sample also.

Reply:
**@v-shalinir**  - No need to remove the C# code sample—just remove the C# code snippets. Since we're also using JavaScript code within the C# sample, please keep the C# code sample.